File: backend/src/utils/DTO's/Request/Pedidos/PedidoCreateDto.js
class PedidoCreateDto {
  /**
   * @param {Object} data
   * @param {number} data.usuarioId
   * @param {string} [data.transportista]
   * @param {number[]} [data.productoIds]
   * @param {number} data.subtotal
   * @param {number} data.flete
   * @param {number} data.iva
   * @param {number} data.total
   */
  constructor(data = {}) {
    this.usuarioId = data.usuarioId ?? 0;
    this.transportista = data.transportista ?? '';
    this.productoIds = data.productoIds ?? [];
    this.subtotal = data.subtotal ?? 0;
    this.flete = data.flete ?? 0;
    this.iva = data.iva ?? 0;
    this.total = data.total ?? 0;
  }
}

export default PedidoCreateDto;

File: backend/src/utils/DTO's/Request/Pedidos/PedidocreateDto.cs
public class PedidoCreateDto
{
    public int UsuarioId { get; set; }
    public string Transportista { get; set; } = string.Empty;
    public List<int> ProductoIds { get; set; } = new();
    public decimal Subtotal { get; set; }
    public decimal Flete { get; set; }
    public decimal IVA { get; set; }
    public decimal Total { get; set; }
}